src/gui/guiChatConsole.cpp

```diff
@@ -96,12 +96,21 @@ GUIChatConsole::GUIChatConsole(
 	// track ctrl keys for mouse event
 	m_is_ctrl_down = false;
 	m_cache_clickable_chat_weblinks = g_settings->getBool("clickable_chat_weblinks");
+
+	m_scrollbar_enabled = g_settings->getBool("chat_scrollbar");
+	m_scrollbar = new GUIScrollBar(env, this, -1, core::rect<s32>(0, 0, 30, m_height), false, true, tsrc);
```

Comment:
Is a unique pointer not an option?

Reply:
Now I'm using an unique pointer.

Reply:
The reason I don't is that that would not call m_scrollbar->drop, and thus screw up the reference counter.

Reply:
Also, the scrollbar lives as long as the console itself does (aka. as long as the game is running, so not in the main menu), so it only needs to be deleted in one place: the destructor of the console window itself.

Reply:
Right, lovely Irrlicht reference counting via inheritance. I sure hope we will have it refactored away to `std::shared_ptr` one day. Arguably a unique pointer would still work (since you can supply a custom deleter), but indeed doesn't add much value.